File: LeetcodeSolutions/Hashing/LongestConsecutiveSequence.java
import java.util.HashSet;
public class LongestConsecutiveSequence {
    public static void main(String[] args) {
        int[] nums = new int[] {100, 4, 200, 1, 3, 2};
        System.out.println(longestConsecutive(nums));
    }

    public static int longestConsecutive(int[] nums) {
        HashSet<Integer> numSet = new HashSet<>();

        for (int num : nums) numSet.add(num);

        int maxSequenceLength = 0;

        for (int i = 0; i < nums.length; i++) {
            int currentNum = nums[i];
            int currentSequenceLength = 1;

            if (!numSet.contains(currentNum - 1)) {
                while (numSet.contains(currentNum + 1)) {
                    currentNum += 1;
                    currentSequenceLength +=1;
                }

                maxSequenceLength = Math.max(maxSequenceLength, currentSequenceLength);
            }
        }
        return maxSequenceLength;
    }
}

File: LeetcodeSolutions/Hashing/LongestConsecutiveSequence.py
def longestConsecutive(nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        longest = 0
        hashset = set(nums)
        for num in nums:
            #Check to see if it's the first of a sequence
            if num - 1 not in hashset:
                length = 1
                while num + length in hashset:
                    length += 1
                longest = max(longest, length)
        return longest

nums = [100,4,200,1,3,2]
print(longestConsecutive(nums))